process/elasticproc/block/blockProcessor.go

```diff
@@ -131,14 +131,81 @@ func (bp *blockProcessor) PrepareBlockForDB(obh *outport.OutportBlockWithHeader)
 		}
 	}
 
+	for _, executionResult := range obh.Header.GetExecutionResultsHandlers() {
+		elasticBlock.ExecutionResultBlockHashes = append(elasticBlock.ExecutionResultBlockHashes, hex.EncodeToString(executionResult.GetHeaderHash()))
+	}
+
 	bp.addEpochStartInfoForMeta(obh.Header, elasticBlock)
 
-	appendBlockDetailsFromHeaders(elasticBlock, obh.Header, obh.BlockData.Body, obh.TransactionPool)
+	elasticBlock.MiniBlocksDetails = prepareMiniBlockDetails(obh.Header.GetMiniBlockHeaderHandlers(), obh.BlockData.Body, obh.TransactionPool)
+
 	appendBlockDetailsFromIntraShardMbs(elasticBlock, obh.BlockData.IntraShardMiniBlocks, obh.TransactionPool, len(obh.Header.GetMiniBlockHeaderHandlers()))
 
 	addProofs(elasticBlock, obh)
 
-	return elasticBlock, nil
+	executionResultData, err := bp.prepareExecutionResults(obh)
+	if err != nil {
+		return nil, err
+	}
+
+	return &data.PreparedBlockResults{
+		Block:            elasticBlock,
+		ExecutionResults: executionResultData,
+	}, nil
+}
+
+func (bp *blockProcessor) prepareExecutionResults(obh *outport.OutportBlockWithHeader) ([]*data.ExecutionResult, error) {
```

Comment:
There is no error returned in this function

Reply:
refactored